### PR TITLE
Add documentation about PrioritySortingWithinCohort feature

### LINKED ISSUE #1283: Do not consider priorities when sorting workloads from different ClusterQueues
**What would you like to be added**:

Use priorities to sort workloads within a ClusterQueue, but ignore the priorities when sorting the heads of multiple cluster queues within a cohort.

The design needs to avoid the race condition presented here https://github.com/kubernetes-sigs/kueue/pull/1024

**Why is this needed**:

In organizations where teams do not know each other, they might be incentivized to use higher priorities to always be ahead of the rest. Ignoring priorities across ClusterQueues would remove this incentive, while allowing users to use priorities within their ClusterQueue.

Maybe this calls for a setting in a new Cohort object.

**Completion requirements**:

This enhancement requires the following artifacts:

- [x] Design doc
- [x] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:
Fixes #1283

#### Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
@yaroslava-serdiuk Can we also add the explanations for the PrioritySortingWithinCohort?

I believe that we can describe this feature in the https://github.com/kubernetes-sigs/kueue/blob/98ec37470b09a3e11baf3de1f1759500c8af32d9/site/content/en/docs/concepts/cluster_queue.md#cohort.

**Comment by yaroslava-serdiuk** [comment]
@tenzen-y added a note for the feature. I think it's enough, because the feature is a temporary (at some point we will graduate the feature, i.e. the feature is always enabled; you cannot disable it or deprecate the feature if we find another solution for workloads sorting).
cc @alculquicondor

File: site/content/en/docs/concepts/cluster_queue.md [hunk 1]
```diff
@@ -303,6 +303,10 @@ a ClusterQueue can borrow up to the sum of nominal quotas from all the
 ClusterQueues in the cohort. So for the yamls listed above, `team-b-cq` can
 borrow `12+9` CPUs.
 
+**Note:** Whithin a Cohort Kueue prioritizes scheduling workloads that will fit under `nominalQuota`.
```
Comment (alculquicondor) [inline site/content/en/docs/concepts/cluster_queue.md:306]:
```suggestion
**Note:** Whithin a Cohort, Kueue prioritizes scheduling workloads that will fit under `nominalQuota`.
```

Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md:306]: This should probably be moved to  line 196

File: site/content/en/docs/concepts/cluster_queue.md [hunk 2]
```diff
@@ -303,6 +303,10 @@ a ClusterQueue can borrow up to the sum of nominal quotas from all the
...
+If multiple workloads require `borrowing`, Kueue will try to schedule workloads with higher [priority](/docs/concepts/workload#priority) by default.
```
Comment (alculquicondor) [inline site/content/en/docs/concepts/cluster_queue.md:307]:
```suggestion
By default, if multiple workloads require borrowing, Kueue will try to schedule workloads with higher [priority](/docs/concepts/workload#priority) first.
```

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**